geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureSource.scala

```diff
@@ -85,7 +85,14 @@ class GeoMesaFeatureSource(val ds: DataStore with HasGeoMesaStats,
   override def getFeatures(filter: Filter): SimpleFeatureCollection =
     getFeatures(new Query(sft.getTypeName, filter))
 
-  override def getFeatures(query: Query): SimpleFeatureCollection = collection(query, this)
+  override def getFeatures(query: Query): SimpleFeatureCollection = {
+    // GeoServer querylayer extension generates query instances w/o type name set
+    if(query.getTypeName == null) {
+      logger.warn(s"Received Query with null TypeName, setting to ${sft.getTypeName}")
```

Comment:
do we want to log this as a warning, or is it expected behavior?

Reply:
it's expected after looking at [this](https://github.com/geoserver/geoserver/blob/master/src/extension/querylayer/src/main/java/org/geoserver/filter/function/QueryFunction.java#L118) but i don't know if it's typically expected.  no strong opinion on whether to keep or not...